config/config.go

```diff
@@ -106,6 +106,7 @@ type Complement struct {
 	// disable this behaviour being added later, once this has stablised.
 	EnableDirtyRuns bool
 
+	// The hostname that will be used to bind the homeserver ports to, e.g. `127.0.0.1`
 	HSPortBindingIP string
```

Comment:
More pertinently: this doesn't seem to reflect the actual behaviour you've introduced for HSPortBindingIP here. AFAICT in your change, it doesn't change the address we bind to, just the way we build the URLs that we use for reaching Synapse.

Maybe that's fine. TBH I couldn't find any documentation for what `HSPortBindingIP` is actually supposed to do, which is why I gave up and added a TODO over in #779. (It seems to only be set for homerunner? Which I have never tried out, and only have the haziest idea of its purpose. If you have got this far, maybe you know more than me.)

In any case, if we're going to just use it for the URLs, the comment had better say that rather than anything about binding.

Reply:
> More pertinently: this doesn't seem to reflect the actual behaviour you've introduced for HSPortBindingIP here. AFAICT in your change, it doesn't change the address we bind to, just the way we build the URLs that we use for reaching Synapse.

Ideally, we would use it for binding but it's just not compatible with the way we `PublishAllPorts` now. I looked for some config while making this PR but the `HostIp` config is only available with the manual port map. With `PublishAllPorts`, we just bind to all interfaces and just look for a binding match to `HSPortBindingIP`.

Using it directly for binding may still be relevant for other custom `Deployer` but the key thing is that it's at-least listening on the specified IP.

I can add a comment to clarify further in a few places.